### PR TITLE
add inspection for calls to a data class' copy method without named arguments.

### PR DESCRIPTION
**Pull request by cypressious**
Fixes #KT-17660

### CODE REVIEW

**Comment by cypressious** [comment]
@mglukhikh I haven't found a solution for hiding the duplicate intention without using `IntentionBasedInspection`, yet.

**Comment by mglukhikh** [comment]
OK, I'll try to find the solution myself :)

File: idea/src/org/jetbrains/kotlin/idea/inspections/CopyWithoutNamedArgumentsInspection.kt [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+class CopyWithoutNamedArgumentsInspection : AbstractKotlinInspection() {
+
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor {
+        return object : KtVisitorVoid() {
+            override fun visitCallExpression(expression: KtCallExpression) {
+                super.visitCallExpression(expression)
+
+                val reference = expression.referenceExpression() as? KtNameReferenceExpression ?: return
+                if (reference.getReferencedNameAsName() != DataClassDescriptorResolver.COPY_METHOD_NAME) return
+                if (expression.valueArguments.none { !it.isNamed() }) return
```
Comment (mg6maciej) [inline idea/src/org/jetbrains/kotlin/idea/inspections/CopyWithoutNamedArgumentsInspection.kt]: `.all { it.isNamed() }` is easier to read.

File: idea/testData/inspections/copyWithoutNamedArguments/test.kt [hunk 2]
```diff
@@ -0,0 +1,15 @@
+data class Foo(val a: String) {
+    fun copy(i: Int) {}
+}
+
+class Foo2() {
+    fun copy(i: Int) {}
+}
+
```
Comment (mg6maciej) [inline idea/testData/inspections/copyWithoutNamedArguments/test.kt]: I'd add an example that clearly shows all should be named, like this:
```
data class SomeName(a: Int, b: Int, c: String)
f.copy(2, c = "")
```

Comment (mg6maciej) [inline idea/testData/inspections/copyWithoutNamedArguments/test.kt]: And maybe another where `copy` is called on `this`:
```
fun SomeName.func() = copy(0)
```

**Comment by mglukhikh** [comment]
I'll improve this inspection and merge it into master myself.

**Pull request closed without merging**